### PR TITLE
BZ-1683087: Corrected IDP data types and included secret/configmap

### PR DESCRIPTION
**Pull request by huffmanca**
BZ-1683087: Corrected IDP data types and included secret/configmap instruction.

This is for OS 4.0.

### CODE REVIEW

**Comment by huffmanca** [comment]
@barleyer ,

I've corrected the data types and included the Secret and ConfigMap instructions in all of the assemblies where it is used within the CR. Note that this means some assemblies, like the HTPasswd instructions, do not contain details on defining a ConfigMap.

Let me know your thoughts, or if you have any questions on these instructions.

File: modules/identity-provider-config-map.adoc [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+= Creating a ConfigMap
+
+Identity providers use {product-title} ConfigMaps to contain the 
+certificate authority bundle.  
+
+* Define an {product-title} ConfigMap containing a the
+certificate authority by using the following command.
++
+----
+$ oc client configmap ca-config-map --from-file=/path/to/ca
```
Comment (barleyer) [inline modules/identity-provider-config-map.adoc]: oc client? maybe a typo! 
And using key **ca.crt**:
`oc create configmap ca-config-map --from-file=ca.crt=/path/to/ca`

**Comment by barleyer** [comment]
@huffmanca hi, many `oc client`,I have no idea it's your desire or a typos.

File: modules/identity-provider-config-map.adoc (hunk 1 above)
Comment (huffmanca) [reply modules/identity-provider-config-map.adoc]: The `oc client` commands have been updated to say `oc create`. In addition, the ConfigMap command now uses your text.

The PR has also been rebased.

**Comment by huffmanca** [comment]
@barleyer,

These issues have been addressed, and this should be ready for review once more.

**Pull request merged**